source/Nevermore/Advanced/ReadTransaction.cs

```diff
@@ -66,19 +66,23 @@ public ReadTransaction(IRelationalStore store, RelationalTransactionRegistry reg
 
         protected DbTransaction? Transaction { get; private set; }
 
+        TimedSection? TransactionTimer { get; set; }
+
         public void Open()
         {
             if (!configuration.AllowSynchronousOperations)
                 throw new SynchronousOperationsDisabledException();
 
             connection = new SqlConnection(registry.ConnectionString);
             connection.OpenWithRetry();
+            TransactionTimer = new TimedSection(ms => configuration.TransactionLogger.Write(ms, name));
         }
 
         public async Task OpenAsync()
         {
             connection = new SqlConnection(registry.ConnectionString);
             await connection.OpenWithRetryAsync().ConfigureAwait(false);
+            TransactionTimer = new TimedSection(ms => configuration.TransactionLogger.Write(ms, name));
```

Comment:
I was thinking of just using Stopwatch as `TimedSection` was designed to use with `using` and a single method, however it looks like that it's used in this way for query logging, so better stick to that pattern.